### PR TITLE
Explain the signature of FrameInterface

### PR DESCRIPTION
**Pull request by mnapoli**
This documentation is a transcript of the discussions in https://groups.google.com/d/topic/php-fig/V12AAcT_SxE/discussion and #755

### CODE REVIEW

File: proposed/http-middleware/middleware-meta.md [hunk 1]
```diff
@@ -171,37 +171,54 @@ that are based around StackPHP.
...
+The `$next` argument is a callable in most existing middleware systems. However using
+an interface allows to improve type safety and IDE support.
+
+Using `__invoke()` as the method name was considered but was not kept. That solution
+would have had the benefit of keeping the existing practice of invoking `$next`
+directly (`$response = $next($request)`). However many existing middleware systems
+already use a class that implements `__invoke()` for the `$next` argument with
+a different signature: `$request` *and* `$response`. Using the same method name
+has the potential to make PSR-15 completely incompatible with these existing
+systems without major changes.
```
Comment (weierophinney) [inline proposed/http-middleware/middleware-meta.md]: I'd maybe rephrase this last sentence to:

> By selecting a different method, PSR-15 may be implemented within existing systems in parallel with existing features, providing a migration path for these libraries and their consumers. In some cases, existing `__invoke()` implementations could even proxy to the method defined by PSR-15, or vice versa.

Comment (mnapoli) [inline proposed/http-middleware/middleware-meta.md]: :+1: updated

**Comment by pmjones** [comment]
Let me know when you feel it's ready and I'll approve.

**Comment by shadowhand** [comment]
:+1:
